engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala

```diff
@@ -26,16 +35,29 @@ class GcsBatchFlowSpec extends AnyFlatSpecLike with CromwellTimeoutSpec with Mat
       } yield new StorageException(code, String.format(ErrorTemplate, op, UnreadableBucketName))
     }
     // Can't flatMap this since any Nones would just be squashed out.
-    storageExceptions(objectReadOperationNames) map GcsBatchFlow.getReadForbiddenBucket map { _.get } shouldBe Set(UnreadableBucketName)
+    storageExceptions(objectReadOperationNames) map (_.getMessage) map GcsBatchFlow.getReadForbiddenBucket map { _.get } shouldBe Set(UnreadableBucketName)
 
     // A sampling of write operations, not an exhaustive list.
     val objectWriteOperationNames = Set(
       "compose",
       "delete",
       "insert"
     )
-    storageExceptions(objectWriteOperationNames) flatMap GcsBatchFlow.getReadForbiddenBucket shouldBe Set.empty
+    storageExceptions(objectWriteOperationNames) map (_.getMessage) flatMap GcsBatchFlow.getReadForbiddenBucket shouldBe Set.empty
+
+    Set(new RuntimeException("random exception")) map (_.getMessage) flatMap GcsBatchFlow.getReadForbiddenBucket shouldBe Set.empty
+  }
+
+  "GcsBatchFlow" should "not throw unhandled exception and kill the thread when trying to recover from unretryable exception with null error message" in {
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+    val gcsBatchFlow = new GcsBatchFlow(1, system.scheduler, null, "testAppName")
+
+    val mockGcsPath = GcsPath(CloudStorageFileSystem.forBucket("bucket").getPath("test"), any[com.google.api.services.storage.Storage], any[com.google.cloud.storage.Storage], anyString)
+    val gcsBatchCommandContext = GcsBatchCommandContext(GcsBatchCrc32Command(mockGcsPath), TestProbe().ref)
+    val recoverCommandPrivateMethod = PrivateMethod[PartialFunction[Throwable, Future[GcsBatchResponse[_]]]]('recoverCommand)
```

Comment:
this is magic

Reply:
Yeah, I learned this technique recently from Mark's S3 PR